app/src/main/java/com/sklein/ubimo_pager/WelcomeActivity.java

```diff
@@ -24,15 +29,21 @@ public void onClick(View v) {
             }
         });
 
-        checkForPermission(Manifest.permission.RECEIVE_SMS);
-        checkForPermission(Manifest.permission.READ_SMS);
+        checkForPermission(NEEDED_PERMISSIONS);
     }
 
-    private void checkForPermission(String permission) {
-        final int permissionStatus = ContextCompat.checkSelfPermission(this, permission);
-        if (permissionStatus == PackageManager.PERMISSION_GRANTED) {
-            return;
+    private void checkForPermission(String[] permissions) {
+        ArrayList<String> permissionsToAsk = new ArrayList<String>();
+        for (String permission : permissions) {
+            int permissionStatus = ContextCompat.checkSelfPermission(this, permission);
+            if (permissionStatus != PackageManager.PERMISSION_GRANTED) {
+                permissionsToAsk.add(permission);
+            }
+        }
+
+        if (!permissionsToAsk.isEmpty()) {
```

Comment:
Why do you want to ask twice for the permissions ?

Reply:
I'm actually asking them all together

Reply:
Right! my mistake, I thought I saw `requestPermissions` usage in the first loop